### PR TITLE
Send TNT alert in admin chat

### LINKED ISSUE #557: Send admin alert when a TNT is defused
This was a feature previously (probs in OCN version of the code).

Useful for when you need spam defuse a team griefing player's TNT so do not have the opportunity to report.

### PR DESCRIPTION
**Pull request by jacobbearden**
Signed-off-by: Jacob Bearden <jacob@bearden.io>

Closes #557 

image

### CODE REVIEW

File: util/src/main/i18n/templates/moderation.properties [hunk 1]
```diff
@@ -157,6 +157,13 @@ moderation.defuse.world = You defused world TNT.
 # {0} = player name
 moderation.defuse.player = You defused {0}'s TNT.
 
+# {0} = defuser player name
+# {1} = griefer player name
+moderation.defuse.alert.player = {0} defused {1}'s TNT.
```
Comment (Pablete1234) [inline util/src/main/i18n/templates/moderation.properties]:
```suggestion
moderation.defuse.alert.player = {0} defused {1}'s {2}.
```
TNT should be localized aswell, using minecraft's name

File: util/src/main/i18n/templates/moderation.properties [hunk 2]
```diff
@@ -157,6 +157,13 @@ moderation.defuse.world = You defused world TNT.
...
+
+# {0} = defuser player name
+moderation.defuse.alert.world = {0} defused world TNT.
```
Comment (Pablete1234) [inline util/src/main/i18n/templates/moderation.properties]: same as above

File: core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java [hunk 3]
```diff
@@ -90,12 +90,25 @@ private void participantDefuse(Player player, Entity entity) {
                     "moderation.defuse.player",
                     clicker.getBukkit(),
                     owner.getBukkit().getDisplayName(clicker.getBukkit()) + ChatColor.RED));
+
+        ChatDispatcher.broadcastAdminChatMessage(
+            TranslatableComponent.of(
+                "moderation.defuse.alert.player",
+                TextColor.WHITE,
+                clicker.getName(NameStyle.FANCY),
+                owner.getName(NameStyle.FANCY)),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java]:
```suggestion
                owner.getName(NameStyle.FANCY),
                MinecraftTranslations.getEntity(entity.getType()).color(TextColor.DARK_RED)),
```
Add the translatable component for the TNT name

File: core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java [hunk 4]
```diff
@@ -90,12 +90,25 @@ private void participantDefuse(Player player, Entity entity) {
...
       } else {
         this.notifyDefuse(
             clicker,
             entity,
             ChatColor.RED
                 + TextTranslations.translate("moderation.defuse.world", clicker.getBukkit()));
+
+        ChatDispatcher.broadcastAdminChatMessage(
+            TranslatableComponent.of(
+                "moderation.defuse.alert.world", TextColor.WHITE, clicker.getName(NameStyle.FANCY)),
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java]: same as above

**Review by Pablete1234 — CHANGES_REQUESTED** [review]

**Comment by Pugzy** [comment]
Not a clue why I didn't PR this [Pugzy/defuse-alert](https://github.com/Pugzy/PGM/tree/defuse-alert) beat me to the punch! 
Just don't quit after leaving the comments above unresolved for a month.

File: core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java [hunk 5]
```diff
@@ -90,12 +90,25 @@ private void participantDefuse(Player player, Entity entity) {
                     "moderation.defuse.player",
                     clicker.getBukkit(),
                     owner.getBukkit().getDisplayName(clicker.getBukkit()) + ChatColor.RED));
+
+        ChatDispatcher.broadcastAdminChatMessage(
+            TranslatableComponent.of(
+                "moderation.defuse.alert.player",
+                TextColor.WHITE,
```
Comment (applenick) [inline core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java]:
```suggestion
                TextColor.GRAY,
```
Consider the gray chat color so formatting is more aligned with other admin broadcasts (e.g /sn) 🙂

**Review by applenick — CHANGES_REQUESTED** [review]

File: core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java (hunk 4 above)
Comment (applenick) [inline core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java]: See other comment about color

**Review by applenick — APPROVED** [review]

**Review by Pablete1234 — APPROVED** [review]

**Review by Electroid — APPROVED** [review]

**Pull request merged**